landing-pages/src/js/integrationList.js

```diff
@@ -0,0 +1,153 @@
+const templateElement = document.createElement("div");
+templateElement.innerHTML = templateText;
+
+const loadingIndicator = root.querySelector(".loading");
+const listItems = root.querySelector(".list-items");
+const searchBox = root.querySelector('[type="search"]');
+const moreButton = root.querySelector("#show-more-integration");
+
+let currentPage = 1;
+let currentQeury = "";
+function setIndicatorVisibility(visible) {
+  loadingIndicator.style.display = visible ? "" : "none";
+  listItems.style.display = visible ? "none" : "";
+}
+setIndicatorVisibility(false);
+function setMoreButtonVisibility(visible) {
+  moreButton.style.display = visible ? "" : "none";
+}
+setMoreButtonVisibility(true);
+
+const sortByLogoAvailability = (a, b) => {
+  const a_key = a.logo ? 1 : -1;
+  const b_key = b.logo ? 1 : -1;
+  return a_key - b_key;
+};
+
+const sortByName  = (a, b) => {
+  return a.name.toLowerCase().localeCompare(b.name.toLowerCase());
+};
+
+const sortByIndex = (a, b) => {
+  return a.index - b.index;
+};
+
+let fetchIntegrationRequest = null;
+
+function fetchIntegration() {
+  if (fetchIntegrationRequest) {
+    return fetchIntegrationRequest;
+  }
+  const request = Promise.all([
+    fetch("/integrations.json"),
+    () => {
+      setIndicatorVisibility(true);
+    }
+  ])
+    .then(([resp]) => resp.json())
+    .then((integrations) => {
+      setIndicatorVisibility(false);
+      return Promise.resolve(integrations);
+    })
+    .then((integrations) => {
+      integrations = shuffle(integrations);
+      integrations.forEach((i, index) => i.inddex = index);
+      integrations.sort(sortByLogoAvailability);
+      return Promise.resolve(integrations);
+    });
+  fetchIntegrationRequest = request;
+  return request;
+}
+
+function createElement(item) {
+  const element = templateElement.cloneNode(true);
+  element.querySelector('[data-name="name"]').innerText = item.name;
+  element.querySelector("a").href = item.url;
+  return element.firstElementChild;
+}
+
+function setItems(items) {
+  if (items.length === 0) {
+    listItems.innerText = "No items";
+  } else {
+    while (listItems.firstChild) {
+      listItems.removeChild(listItems.firstChild);
+    }
+    items.forEach((item) => {
+      const element = createElement(item);
+      listItems.append(element);
+    });
+  }
+}
+
+function showItems(keyword, page) {
+  const showMoreButtonIfNeeded = (integrations) => {
+    setMoreButtonVisibility(integrations.length > (page * 8));
+  };
+
+  const filterMatchingItems = (integrations) => {
+    if (!keyword) {
+      return Promise.resolve(integrations);
+    }
+    const selectedIntegration = integrations.filter(
+      (integration) => integration.name.indexOf(keyword) >= 0
+    );
+    return Promise.resolve(selectedIntegration);
+  };
+
+  const filterVisible = (integrations) => {
+    if (currentPage === 1) {
+      integrations.sort(sortByIndex);
+    } else {
+      integrations.sort(sortByName);
```

Comment:
Do we want to have different order of items after clicking the "show more" button? Isn't it a bit confusing?

Reply:
I don't know the solution, but it's actually confusing, but posting a random list will also always be misleading, and it's necessary not to promote one integration.

Reply:
I agree, but we achieve randomness by shuffling integrations on line 79. What I mean here is maybe we shouldn't change the type of sorting after clicking button (so it would just be sorted by index).

Reply:
Viewing a long random list will be more confusing than reordering a few items after clicking, but I can change that and ask for feedback tomorrow.